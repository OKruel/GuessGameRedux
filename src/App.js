import React, { Component } from 'react';
import { connect } from 'react-redux'

import './App.css';
import GuessedWords from './GuessedWords'
import Congrats from './Congrats'
import Input from './Input'
import GuessCount from './GuessCount'
import NewWord from './NewWord'

import { getSecretWord } from './actions/index'

export class UnconnectedApp extends Component {
  componentDidMount(){
    this.props.getSecretWord()
  }
  
  render() {
    return (
      <div className='container'>
        <h1>Guess Word Game</h1>
        <div>The secret word is: {this.props.secretWord}</div>
        <Congrats success={this.props.success} />
        {this.props.success?<NewWord/>:<></> }
        <Input/>
        <GuessedWords guessedWords={this.props.guessedWords} />
        {this.props.guessedWords.length > 0? <GuessCount ></GuessCount>: <></> }
      </div>
    );
  }
}

const mapStateToProps = state => {
  const {success, guessedWords, secretWord} = state
  return {success, guessedWords, secretWord}
}

export default connect(mapStateToProps, { getSecretWord })(UnconnectedApp);
