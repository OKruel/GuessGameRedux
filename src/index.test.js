import React from 'react'
import { shallow } from 'enzyme'

import App from './App'
import { Provider } from 'react-redux'
import store from './configStore'

test('renders propertly', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
})

test('were created with the root div', () => {
    const props = {store}
    const wrapper = shallow(<Provider {...props}/>)
    expect(wrapper).toMatchSnapshot()
})