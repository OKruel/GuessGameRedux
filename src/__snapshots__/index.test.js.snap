// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders propertly 1`] = `ShallowWrapper {}`;

exports[`were created with the root div 1`] = `ShallowWrapper {}`;
